### DISCUSSION TITLE
MatrixBlockQuery::customFields check for selected type Id

### DISCUSSION THREAD

**Discussion by Anubarak** [craftcms/cms#9921] [Ideas] [answered]
Just an idea:
I guess it would be useful to check the current selected typeId when the MatrixQuery returns its `customFields`
Imagine you have a Block with two different types and both share a `headline` field

```PHP
$entry->getFieldValue('matrixField')
    ->typeId('X')
    ->headline(':notempty:')
    ->one();
```
will always return `null` because both headline fields are will check for `notempty` but `content.field_blockX_headline` will always be `null` 

If you could check for the current `typeId` in the `customFields` functions and only return those fields from the matching block types your could actually filter for matrix blocks - right now all blocks must "pass" the filter condition.

Steps to reproduce:
1) create a matrix field
2) add two block types with the very same field handles
3) try to filter a specific block with a type and field value without modifying the query per hand via `where` conditions

**Comment by brandonkelly** [comment, marked as answer]
Great idea. Just released Craft 3.7.15 with that change (0438530c3b5).
